lib/widget/answer_multiple_choices.dart

```diff
@@ -0,0 +1,122 @@
+  final SelectionAnswerType answerType;
+
+  const AnswerMultipleChoices({
+    super.key,
+    required this.answers,
+    required this.answerType,
+  });
+
+  @override
+  State<AnswerMultipleChoices> createState() => _AnswerMultipleChoicesState();
+}
+
+class _AnswerMultipleChoicesState extends State<AnswerMultipleChoices> {
+  final List<String> _selectedAnswerIds = [];
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: [
+        for (int index = 0; index < widget.answers.length; index++) ...[
+          _buildItem(
+            widget.answers[index],
+            _selectedAnswerIds.contains(widget.answers[index].id),
+          ),
+          if (index < widget.answers.length - 1) _buildItemDivider(),
+        ],
+      ],
+    );
+  }
+
+  Widget _buildItem(SurveyAnswerUiModel answer, bool isSelectedAnswer) {
+    return GestureDetector(
+      onTap: () {
+        setState(() {
+          _buildAnswerSelection(answer.id, !isSelectedAnswer);
+        });
+      },
+      child: Container(
+        height: multipleChoiceItemHeight,
+        alignment: Alignment.center,
+        child: Row(
+          children: [
+            _buildItemText(answer, isSelectedAnswer),
+            _buildItemCheckBox(answer, isSelectedAnswer),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildItemText(SurveyAnswerUiModel answer, bool isSelectedAnswer) {
+    return Expanded(
+      child: Text(
+        answer.text,
+        style: TextStyle(
+          color:
+              isSelectedAnswer ? Colors.white : Colors.white.withOpacity(0.5),
+          fontSize: fontSize20,
+          fontWeight: isSelectedAnswer ? FontWeight.w800 : FontWeight.w400,
+        ),
+      ),
+    );
+  }
+
+  Widget _buildItemCheckBox(SurveyAnswerUiModel answer, bool isSelectedAnswer) {
+    Color getColor(Set<MaterialState> states) {
+      const Set<MaterialState> interactiveStates = <MaterialState>{
+        MaterialState.pressed,
+        MaterialState.hovered,
+        MaterialState.focused,
+      };
+      if (states.any(interactiveStates.contains)) {
+        return Colors.white.withOpacity(0.5);
+      }
+      return isSelectedAnswer ? Colors.white : Colors.white.withOpacity(0.5);
+    }
+
+    return Checkbox(
+      checkColor: ColorName.gray,
+      fillColor: MaterialStateProperty.resolveWith(getColor),
+      value: _selectedAnswerIds.contains(answer.id),
+      shape: const CircleBorder(),
+      side: BorderSide(
+        color: isSelectedAnswer ? Colors.white : Colors.white.withOpacity(0.5),
+      ),
+      materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
+      onChanged: (bool? value) {
+        setState(() {
+          _buildAnswerSelection(answer.id, value ?? false);
+        });
+      },
+    );
+  }
+
+  void _buildAnswerSelection(String answerId, bool value) {
```

Comment:
I think the function name doesn't reflect its purpose correctly since we don't build any answer selection widget here. How about `_onAnswerSelected()`?

Reply:
@manh-t I think we could name `_onAnswerClicked` because the item can be selected and unselected 😄 And the param can be `value => isSelected`